### PR TITLE
Update multiple winners

### PR DESCRIPTION
**Pull request by asharma13524**
This pr adds multiple options for our next lottery in which we will select three winners.
Option 1:

Call draw_winner multiple times until we achieve 3 unique addresses.

- Adds draw_winner that selects a winner and checks that the address has not won before.

Option 2:

Call draw_three_winners until we achieve 3 unique addresses.

- Adds draw_three_winners (attempts to fill 3 winner slots iff addresses are unique)
- Adds send_three_prizes_to_winners (sends all three prizes to each winner and checks that the addresses are the addresses selected from draw_three_winners). Also takes in the idx at which the winners were set on each respective winner mapping.

Additionally, this pr fixes and adds back in the send_prize_to_winner finalize to check that the invocation of send_prize_to_winner includes the address that has officially won.

This adds a parameter to send_prize_to_winner (the key (idx u32) at which the winner_address you are checking for exists in the winner mapping).

****Note:** We decided on option 1 for now.**

### CODE REVIEW

**Comment by lukenewman** [comment]
we could also use the `has_won` mapping inside `send_prize_to_winner` and not have to pass in the index, right?

**Comment by lukenewman** [comment]
i think i'm leaning towards Option 1 -- the basic flow is just "call `draw_winner` until `winner_count` is 2u32" right?

File: puzzle_raffle/src/main.leo [hunk 1]
```diff
@@ -403,18 +457,83 @@ program puzzle_raffle_050124.aleo {
...
+    finalize send_three_prizes_to_winners (
+        public winner_addr_one: address,
+        public winner_addr_two: address,
+        public winner_addr_three: address,
+        public winner_idx: u32,
+    )
+    {
+        // assert that the prizes are being sent to the official winners
+        assert_eq(Mapping::get(winner_one, winner_idx), winner_addr_one);
+        assert_eq(Mapping::get(winner_one, winner_idx), winner_addr_two);
```
Comment (DarvishKamaliaDL) [inline puzzle_raffle/src/main.leo]: the mappings here should be `winner_two` and `winner_three` yeah?

**Comment by DarvishKamaliaDL** [comment]
yeah option 1 seems simpler, although option two keeps more of the winner selection logic on chain which might be more valuable for transparency / branding purposes.

**Comment by asharma13524** [comment]
> we could also use the `has_won` mapping inside `send_prize_to_winner` and not have to pass in the index, right?

the first winner would actually be 1u32 since we increment first. so up to 3u32 but that's assuming each call selects a winning unique address (which isn't guaranteed).

**Review by lukenewman — APPROVED** [review]

**Pull request merged**